src/System.Runtime.Serialization.Formatters/tests/SerializationGuardTests.cs

```diff
@@ -0,0 +1,178 @@
+        }
+
+        [Fact]
+        void BlockFileWrites()
+        {
+            TryPayload(new FileWriter());
+        }
+
+        [Fact]
+        void BlockReflectionDodging()
+        {
+            MemoryStream ms = new MemoryStream();
+            BinaryFormatter writer = new BinaryFormatter();
+            writer.Serialize(ms, new ReflectionDodger());
+            ms.Position = 0;
+
+            BinaryFormatter reader = new BinaryFormatter();
+            Assert.Throws<TargetInvocationException>(() => reader.Deserialize(ms));
+        }
+
+        [Fact]
+        void BlockAsyncDodging()
+        {
+            TryPayload(new AsyncDodger());
+        }
+
+        static void TryPayload(object payload)
+        {
+            MemoryStream ms = new MemoryStream();
+            BinaryFormatter writer = new BinaryFormatter();
+            writer.Serialize(ms, payload);
+            ms.Position = 0;
+
+            BinaryFormatter reader = new BinaryFormatter();
+            Assert.Throws<DeserializationBlockedException>(() => reader.Deserialize(ms)); // This should really look for DeserializationBlockedException, but it's not in all of the frameworks this test builds for
+        }
+    }
+
+    [Serializable]
+    class AssemblyLoader : ISerializable
+    {
+        public AssemblyLoader() { }
+
+        public AssemblyLoader(SerializationInfo info, StreamingContext context)
+        {
+            Assembly.Load(new byte[1000]);
+        }
+        
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+        }
+    }
+
+    [Serializable]
+    class ProcessStarter : ISerializable
+    {
+        public ProcessStarter() { }
+
+        private ProcessStarter(SerializationInfo info, StreamingContext context)
+        {
+            Process.Start("calc.exe");
+        }
+        
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+        }
+    }
+
+    [Serializable]
+    class FileWriter : ISerializable
+    {
+        public FileWriter() { }
+
+        private FileWriter(SerializationInfo info, StreamingContext context)
+        {
+            string tempPath = Path.GetTempFileName();
+            File.WriteAllText(tempPath, "foo");
+        }
+        
+        public void GetObjectData(SerializationInfo info, StreamingContext context)
+        {
+        }
+    }
+
+    [Serializable]
+    class ReflectionDodger : ISerializable
+    {
+        public ReflectionDodger() { }
+
+        private ReflectionDodger(SerializationInfo info, StreamingContext context)
+        {
+            object tracker = null;
+            Type threadType = typeof(object).Assembly.GetType("xSystem.Threading.Thread");
+            MethodInfo trackerMethod = null;
+            if (threadType != null)
+            {
+                trackerMethod = threadType.GetMethod("GetThreadDeserializationTracker", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static);
```

Comment:
Meetings invoked could be anything right? Maybe something basic like String.Concat might be clearer?